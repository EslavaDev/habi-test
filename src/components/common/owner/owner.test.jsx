import { cleanup, render } from '@testing-library/react';
import * as React from 'react';
import { Owner } from '.';

let props = null;

describe('Owner', () => {
  beforeEach(() => {
    props = {
      fullname: 'carlos',
      phone: '1111111',
      email: 'carlos@mail.com',
    };
  });

  afterEach(() => {
    cleanup();
  });

  test('Should render the component', () => {
    const { container } = render(<Owner {...props} />);
    expect(container).toMatchSnapshot();
  });

  test('Should render the correct fields', () => {
    const { getByTestId } = render(<Owner {...props} />);

    const fullName = getByTestId('owner-name');
    const phone = getByTestId('owner-phone');
    const contact = getByTestId('owner-contact');

    expect(fullName).toBeTruthy();
    expect(phone).toBeTruthy();
    expect(contact).toBeTruthy();
  });
  test('Should render the correct fields by text', () => {
    const { getByText } = render(<Owner {...props} />);

    const fullName = getByText('carlos');
    const phone = getByText('1111111');
    const contact = getByText('carlos@mail.com');

    expect(fullName).toBeTruthy();
    expect(phone).toBeTruthy();
    expect(contact).toBeTruthy();
  });
});
