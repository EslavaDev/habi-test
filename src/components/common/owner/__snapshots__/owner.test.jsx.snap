// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Owner Should render the component 1`] = `
<div>
  <div>
    <p
      data-testid="owner-name"
    >
      <strong>
        Name: 
      </strong>
       
      carlos
    </p>
    <p
      data-testid="owner-phone"
    >
      <strong>
        Phone: 
      </strong>
       
      1111111
    </p>
    <p
      data-testid="owner-contact"
    >
      <strong>
        Contact: 
      </strong>
       
      carlos@mail.com
    </p>
  </div>
</div>
`;
